packages/web-components/src/components/menu/menu-item.ts

```diff
@@ -0,0 +1,339 @@
+  @property()
+  onKeyDown?: (e: KeyboardEvent) => void;
+
+  @property({ attribute: 'aria-checked' })
+
+  /**
+   * Provide an optional function to be called when the MenuItem is clicked.
+   */
+  @property()
+  onClick?: (event: KeyboardEvent | MouseEvent) => void;
+
+  checkForIcon() {
+    if (this.renderIcon) {
+      //this.context.updateFromChild?.({ hasIcons: true },this);
+      // if(!this.context.hasIcons)
+      this.parentElement?.shadowRoot
+        ?.querySelector(`.${prefix}--menu`)
+        ?.classList.add(`${prefix}--menu--with-icons`);
+    } else {
+      // this.hasIcon=false;
+    }
+  }
+
+  firstUpdated() {
+    this.checkForIcon();
+    this.hasChildren = this.childNodes.length > 0;
+    this.isDisabled = this.disabled && !this.hasChildren;
+    this.direction = document.dir;
+    this.isRtl = this.direction === 'rtl';
+    this.isDanger = this.kind === 'danger';
+    setTimeout(() => {
+      this.childElements = Object.values(this.children);
+    }, 100);
+    this._registerSubMenuItems();
+  }
+
+  render() {
+    const {
+      label,
+      shortcut,
+      renderIcon,
+      isDisabled,
+      hasChildren,
+      submenuOpen,
+      _handleClick: handleClick,
+      _handleMouseEnter: handleMouseEnter,
+      _handleMouseLeave: handleMouseLeave,
+      _handleKeyDown: handleKeyDown,
+      _closeSubmenu: closeSubmenu,
+      isDanger,
+      boundaries,
+      childElements,
+      isRtl,
+    } = this;
+    const menuItemClasses = classMap({
+      [`${prefix}--menu-item`]: true,
+      [`${prefix}--menu-item--disabled`]: isDisabled,
+      [`${prefix}--menu-item--danger`]: isDanger,
+      // [`${prefix}--menu-item--with-icon`]:!this.hasIcon
+    });
+    return html`
+      <li
+        part="menuitem"
+        role="menuitem"
+        tabindex="0"
+        class="${menuItemClasses}"
+        aria-disabled="${isDisabled ?? undefined}"
+        aria-haspopup="${hasChildren ?? undefined}"
+        aria-expanded="${hasChildren ? submenuOpen : undefined}"
+        @click="${handleClick}"
+        @mouseenter="${hasChildren ? handleMouseEnter : undefined}"
+        @mouseleave="${hasChildren ? handleMouseLeave : undefined}"
+        @keydown="${handleKeyDown}">
+        <div className="${prefix}--menu-item__selection-icon">
+          ${this.getAttribute('aria-checked') === 'true'
+            ? Checkmark16()
+            : undefined}
+        </div>
+
+        <div class="${prefix}--menu-item__icon">
+          ${renderIcon ? html`${renderIcon()}` : html``}
+        </div>
+        <div class="${prefix}--menu-item__label">${label}</div>
+        ${shortcut &&
+        !hasChildren &&
+        html` <div class="${prefix}--menu-item__shortcut">${shortcut}</div> `}
+        ${hasChildren
+          ? html`
+              <div class="${prefix}--menu-item__shortcut">
+                ${isRtl ? CaretLeft16() : CaretRight16()}
+              </div>
+              <cds-menu
+                .isChild="${hasChildren}"
```

Comment:
If `hasChildren` is a boolean we should be able to apply the attribute or not based on its value

```suggestion
                ?isChild="${hasChildren}"
```

Reply:
Committed to the [latest PR](https://github.com/carbon-design-system/carbon/pull/18718)